applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -401,16 +424,59 @@ static void pgps_event_handler(struct nrf_cloud_pgps_event *event)
 
 #endif /* CONFIG_SLM_NRF_CLOUD */
 
-#if defined(CONFIG_SLM_LOG_LEVEL_DBG)
+#if CONFIG_SLM_GNSS_OUTPUT_NMEA_ON_CMUX_CHANNEL
+static void gnss_nmea_sender(struct k_work *)
+{
+	int ret;
+	size_t size;
+	uint8_t *data;
+
+	while ((size = ring_buf_get_claim(&gnss_nmea_rb, &data, UINT32_MAX))) {
+		ret = modem_pipe_transmit(gnss_pipe, data, size);
```

Comment:
I think that this will deadlock if the pipe transmission keeps returning error. Should we exit with some error codes? Or is this not recoverable anyway?